Content.Shared/Engineering/Components/DisassembleOnAltVerbComponent.cs

```diff
@@ -0,0 +1,25 @@
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Content.Shared.DoAfter;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Engineering.Components;
+
+[RegisterComponent, NetworkedComponent]
+public sealed partial class DisassembleOnAltVerbComponent : Component
+{
+    /// <summary>
+    ///     The prototype that is spawned after disassembly. If null, nothing will spawn.
+    /// </summary>
+    [DataField]
+    public ProtoId<EntityPrototype>? PrototypeToSpawn;
```

Comment:
```suggestion
    public EntProtoId? PrototypeToSpawn;
```
`ProtoId<EntityPrototype>` will cause debug asserts. Maybe that was one of the problems you had?